python/cugraph/structure/graph.py

```diff
@@ -72,7 +72,8 @@ def __init__(self, m_graph=None, edge_attr=None, symmetrized=False,
         """
         self.symmetrized = symmetrized
         self.renumbered = False
-        self.bipartite = bipartite
+        self.bipartite = False
+        self._nodes = {}
```

Comment:
shouldn't nodes be an array of sets?  You need to know which nodes are in sets A and set B.   We should also allow for N-partite

Reply:
@Iroy30 can answer more definitively.  I believe that nodes is a map (python dictionary) with the key being the node type and the value being a list of nodes that are of that type.

Reply:
Updated the API to allow for multipartite graphs. The nodes are a dictionary mapping partition name to the set of nodes in that partition

Reply:
I am okay starting with a map, but also think that it will need to change as we flesh out use cases

the most likely case is that the data will start in a cudf dataframe where there is a column for vertex ID and a column for vertex type.  Creating a python dictionary from that data sounds very inefficient

Reply:
If we look at some work we did with customers.  each partite layer had its own numbering scheme, so there can be collisions

Reply:
If I understand you correctly, on the Dataframe input having a column of vertex IDs and a column of partitionID/name, a cudf groupby could be used to store the map. However, I am not completely clear about the:

> If we look at some work we did with customers. each partite layer had its own numbering scheme, so there can be collisions

Maybe an example usecase would be more helpful in understanding better.

Reply:
vertices in set A = [0,1,2,3,4], vertices in set B = [0,1,2,3, 4]  where A != B.  So the vertex with ID 0 in A is not the same as the vertex in B with ID 0

Reply:
How would the user input of the cudf DataFrame containing the edgelist look like for a Graph() structure in this case.